README-advanced.md

```diff
@@ -90,77 +90,166 @@ the `notebook` portion of repository demonstrating usage of the pipeline.
   
-### Ephys Files
-
-The ephys ingest logic searches the `ephys_data_paths` for processed ephys
-data files. These are specified in dj.config.json as:
-
-    "ephys_data_paths": [["/Users/chris/src/djc/map/map-sharing/unittest/ephys", "0"]]
-
-if this variable is not configured, hardcoded values in the code matching
-the experimental setup will be used.
-
-File paths conform to the pattern:
+#### Ephys Files Ingestion
+Ephys Files are typically structured in a directory convention as followed:
 
-    \2017-10-21-1\tw5ap_imec3_opt3_jrc.mat
+    {h2o}/{YYYYMMDD}/{probe_no}/ephys_file  
+or
 
-which is, more generally:
-
-    \{YYYY}-{MM}-{DD}-{N}\{h2o}_ap_imec3_opt3_jrc.mat
-
-Where:
-
-    - YYYY: 4-Digit Year
-    - MM: 2-Digit Month
-    - DD: 2-Digit Month
-    - N: Probe number for this probe
-
-Older files matched the pattern:
-
-    {h2o}_g0_t0.imec.ap_imec3_opt3_jrc.mat
-    {h2o}_g0_t0.imec.ap_imec3_opt3_jrc.mat
+    {h20}/{h20}_{YYYYMMDD}_*/{h20}_{YYYYMMDD}_*_imec[0-9]/
+    
+Ephys Files can be:
++ JRClust v3 - e.g. "tw5ap_imec3_opt3_jrc.mat"
+    + `{h2o}_*_jrc.mat`
++ JRClust v4 - e.g. "tw5ap_imec3_opt3.ap_res.mat"
+    + `{h2o}_*.ap_res.mat`
++ Kilosort2 - *ks* folder with a set of ks output files (e.g. spike_times.npy, template.npy, etc.)
+    
+Some example ephys paths:
+    
+    dl40/20181022/1/dl40_g0_t50.imec.ap_imec3_opt3_jrc.mat
+    SC022/20190303/1/SC022_g0_t0.imec0.ap_res.mat
+    SC035/catgt_SC035_010720_g0/SC035_010720_g0_imec0/imec0_ks2
+    SC035/catgt_SC035_011020_g0/SC035_011020_g0_imec2/imec2_ks2_orig
+    
+### Manual session-based ephys ingest
+This section introduces additional manual routines for ephys ingest for any given behavior session after the main ephys ingestion (i.e. `mapshell.py ingest-ephys`) has been completed. 
+There are 2 reasons for performing manual ephys ingestion after the `mapshell.py ingest-ephys` routine:    
+
+1. Extending an ingested set of ephys results
+    + This is needed when not all data were available during the 1st ingestion run. For instance, clustering results for probe 3 was not available at the time of the 1st ingestion - thus only 2 probe ephys data were ingested in a 3-probe recording session
+    + This does not modify the already ingested ephys data, only extending it
+2. Replacing an ingested set of ephys results
+    + This is needed for a new version of curated clustering results to replace the ingested set of ephys results (or when quality controlled results become available)
+    + There are 4 major steps taken in this routine:
+        1. Search the specify data directory and verify the presence of new clustering results for the specified ***session***
+        2. Copy the ingested ephys data over to the ***ArchiveUnit*** table, with all meta info tracked (this archived results will be stored externally on AWS S3 to reduce cost of storage)
+        3. Delete the ingested data
+        4. Re-ingest new version of the clustering results
```

Comment:
should also document the case of archiving when no replacement units are available - this still might be useful when a set of results is 'known bad' but no replacement set is ready but the old results are still wanted for whatever reason

Reply:
just archiving without replacement may trigger reingest of the same "bad" results the next time users run `mapshell.py ingest-ephys` (users will have to remember to delete the "old" result folder). This may potentially introduce confusion, so let's wait until we know there are needs/use-case for this before publicize it

Reply:
ah ok - good point. makes sense.